Source/Scene/Camera.js

```diff
@@ -29,6 +29,30 @@ import CameraFlightPath from "./CameraFlightPath.js";
 import MapMode2D from "./MapMode2D.js";
 import SceneMode from "./SceneMode.js";
 
+//PROPELLER HACK
+function clampSphere(sphere, position, result) {
+  if (
+    !defined(sphere) ||
+    !defined(position) ||
+    !position ||
+    position === Cartesian3.ZERO
+  ) {
+    return position;
+  }
+
+  result = Cartesian3.clone(position, result);
+  if (Cartesian3.distance(result, sphere.center) <= sphere.radius) {
+    return result;
+  }
+  var dirV = new Cartesian3();
```

Comment:
Happy to have verbose names to help others reading the code.  e.g. `directionVector = ...`